cloudify_aws/ec2/__init__.py

```diff
@@ -29,6 +30,10 @@ class EC2Base(AWSResourceBase):
         AWS ELB base interface
     """
     def __init__(self, ctx_node, resource_id=None, client=None, logger=None):
+        if not client:
+            aws_config = ctx_node.properties.get(AWS_CONFIG_PROPERTY, dict())
+            check_region_name(aws_config['region_name'])
+
```

Comment:
Why do we only run this if the `client` is `None`?

Reply:
If we already have client - i suppose that aws already checked credentials.